### ISSUE TITLE
twisted.conch.test.test_knownhosts.KnownHostsDatabaseTests.test_loadFromPath covers too many cases

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5617]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5617|
|Type|defect|
|Created|2012-04-06 11:50:06Z|
|Branch|https://github.com/twisted/twisted/tree/conch-test_loadFromPath-5617|

This test method has a docstring that includes the phrase "a path with six entries in it".  What's special about six?  Why not seven or five?

The test implementation goes on to do much more than verify there are six entries.  This test should be split up to exercise the different code paths it is trying to exercise separately.  A good scope for a single test method is "is an unparseable line represented as an UnparsedEntry instance" or "is an unparseable line that starts with an encryption marker represented as an UnparsedEntry instance".

These assertions are largely in the test method already, they just shouldn't all be in the _same_ test method.  If the test had been factored as 4 or 5 tests instead, it would have been more obvious that a new test method should be added for [#5616](https://github.com/twisted/twisted/issues/5616).

Attachments:

* [known_hosts_tests.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/126/1266c19d7fa1773ccfed130018cf1da900dd04d3/73718df06af4f14e19246f1291226f97f9eb43a5.patch) (4307 bytes) - added by thijs on 2012-04-28 15:44:53Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5617 5617
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_conch_test_loadFromPath_5617 branches/conch-test_loadFromPath-5617
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__tests_easy tests easy
time__1333713006000000 1333713006000000
changetime__1338057067000000 1338057067000000
version__None None
owner__ 
cc__z3p cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

The attached patch splits the tests up as suggested.

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|z3p commented|
|-|-|

(In [34398]) Branching to 'conch-test_loadFromPath-5617'

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|z3p commented|
|-|-|

1.test_verifyUnparseableEncryptionMarker should L{} to UnparsedEntry.

Other comments:

 1. NEWS entry
 1. 80 character lines in test_knownhosts.py

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [34400]) address review comments, refs [#5617](https://github.com/twisted/twisted/issues/5617)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner|
|-|-|

Thanks for the review. I added a misc file and fixed the epytext and lines that exceeded 80 char in test_knownhosts.

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|z3p commented|
|-|-|

(In [34440]) more 80 column lines

Refs [#5617](https://github.com/twisted/twisted/issues/5617)

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|z3p set status to `closed`|
|-|-|

(In [34446]) Merge conch-test_loadFromPath-5617

Author: thijs
Reviewer: z3p
Fixes: [#5617](https://github.com/twisted/twisted/issues/5617)

Refactors t.c.t.test_knownhosts.KnownHostsDatabaseTests.test_loadFromPath into
indivdual unit tests rather than one monolithic test case.

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|z3p commented|
|-|-|

I suppose I should have let thijs merge this one in, but I was already merging other stuff.

Thanks for the patch!
